### PR TITLE
[4.0] webp support in templates

### PR DESCRIPTION
**Pull request by brianteeman**
#31048 provided the library level support for webp images

This PR adds webp to the list of allowed image types for com_templates. It does not have an update sql as that would change user data.

This isnt really anything to test here accept codestyle as using webp images in templates was the test for the libray update so we know it works

adding support to the media manager will be in its own pr (its a bit more complex)

### CODE REVIEW

File: administrator/components/com_templates/config.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
 			name="image_formats"
 			type="text"
 			label="COM_TEMPLATES_CONFIG_IMAGE_LABEL"
-			default="gif,bmp,jpg,jpeg"
+			default="gif,bmp,jpg,jpeg,webp"
```
Comment (Quy) [inline administrator/components/com_templates/config.xml:37]:
```suggestion
			default="gif,bmp,jpg,jpeg,png,webp"
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7f4cfaa3283f390649dbda59dc20fa25ba67f0f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33087">issues.joomla.org/tracker/joomla-cms/33087</a>.</sub>

**Comment by particthistle** [comment]
I have tested this item :white_check_mark: successfully on 7f4cfaa3283f390649dbda59dc20fa25ba67f0f8<br /><br />Code style change. All change really does is allow webp images to be used in com_templates.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33087">issues.joomla.org/tracker/joomla-cms/33087</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33087">issues.joomla.org/tracker/joomla-cms/33087</a>.</sub>

**Comment by SharkyKZ** [comment]
I have tested this item :red_circle: unsuccessfully on 7f4cfaa3283f390649dbda59dc20fa25ba67f0f8<br /><br />webp images are saved as jpgs, transparency is lost.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33087">issues.joomla.org/tracker/joomla-cms/33087</a>.</sub>

**Comment by brianteeman** [comment]
@SharkyKZ good spot - fixed it (I hope please retest)

@quy please remove RTC for now

**Comment by brianteeman** [comment]
To test the resize and cropping of a webp image. Upload an image and resize and/or crop. Then download the image and open in a text editor. If it is still a webp image you will see something like `RIFF" WEBPVP8 ` at the beginning. If it has been saved incorrectly as a jpeg then it will have `<CREATOR: gd-jpeg v1.0 (using IJG JPEG v90), quality = 100`

File: administrator/components/com_templates/config.xml (hunk 1 above)
Comment (particthistle) [inline administrator/components/com_templates/config.xml:37]: Should this list contain png?

default="gif,bmp,jpg,jpeg,png,webp"

**Review by particthistle — COMMENTED** [review]
Testing the actual process this time 🤦🏻‍♂️ 

The change only impacts the default. To actually test if all this works, I've had to go to System > Site Templates > Options and add wepb (and png) to the list.

I note that the default is missing PNG.

**Comment by particthistle** [comment]
In testing, I also found an issue if the extension of the file is upper case.

sample.PNG would upload, but then not be visible in the file tree. Adding GIF,JPG,JPEG,PNG,WEBP to the list of allowed extensions then allowed the uppercase extension to be viewed.

Is it worth adding that now, or is this something to be raised as a separate issue to fix in the uploader so that it makes the extensions lowercase when the files are uploaded?

**Comment by brianteeman** [comment]
Beyond the scope of this PR - please create a separate issue

**Comment by particthistle** [comment]
I have tested this item :white_check_mark: successfully on e64306b7b2343206c49d5c42a7e61f49678dcd60<br /><br />Addition of png and webp extensions allows file types to be uploaded.
(To test on existing build check System > Site Templates > Options to add png,webp to the image extensions list, as it won't apply the default automatically)

Resizing and Cropping a webp image saves the image as a webp image correctly.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33087">issues.joomla.org/tracker/joomla-cms/33087</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on e64306b7b2343206c49d5c42a7e61f49678dcd60<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33087">issues.joomla.org/tracker/joomla-cms/33087</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33087">issues.joomla.org/tracker/joomla-cms/33087</a>.</sub>

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**